test/unit/url.test.js

```diff
@@ -1327,6 +1327,16 @@ describe('Url', function () {
             ]);
         });
 
+        it('should correctly add hasOwnProperty query key to an existing Url instance', function () {
```

Comment:
For context, let's add link to the reported issue.
```suggestion
        // Refer: https://github.com/postmanlabs/postman-app-support/issues/8924
        it('should correctly add hasOwnProperty query key to an existing Url instance', function () {
```

Reply:
About adding an isolate test for `PropertyList.insert` method. I tried adding a test there and could not get that test to fail. It was passing fine.
The reason is, In the `property-list.test.js` file, when making an instance of PropertyList we do not pass any parameters to the constructor, due to this `_postman_listAllowsMultipleValues` is set to false and due to short circuit [this condition](https://github.com/postmanlabs/postman-collection/blob/043e8b1bb95d48ad2854485b051a47ec90d7d02c/lib/collection/property-list.js#L133), it never checks for remaining && condition (`this.reference.hasOwnProperty(index)`) and also never checks members with multiple values key. That's why test will never fail there.

Sure, can change test display message.

Reply:
You can create a `PropertyList` with a custom type having `_postman_propertyAllowsMultipleValues` set to true.
```js
var FakeType,
    list;

FakeType = function (options) {
    this.key = options.key;
    this.value = options.value;
};

FakeType._postman_propertyIndexKey = 'key';
FakeType._postman_propertyAllowsMultipleValues = true;

list = new sdk.PropertyList(FakeType, {}, { key: 'hasOwnProperty', value: 'false' });
list.insert({ key: 'foo', value: 'bar' }); // Uncaught TypeError: this.reference.hasOwnProperty is not a function
```

Reply:
ok, As you have suggested I have tried adding test for `PropertyList.insert()` method [here](https://github.com/postmanlabs/postman-collection/pull/1081/commits/0b7d4078c1a59bd7aa4ebb54a11919f1d72b2c2f#diff-c6d14c63e95734583772196465ecded5) .